packages/snaps-controllers/src/cronjob/CronjobController.ts

```diff
@@ -267,11 +340,126 @@ export class CronjobController extends BaseController<
   }
 
   /**
-   * Unregister all jobs related to the given snapId.
+   * Schedule a background event.
+   *
+   * @param backgroundEventWithoutId - Background event.
+   * @returns An id representing the background event.
+   */
+  scheduleBackgroundEvent(
+    backgroundEventWithoutId: Omit<BackgroundEvent, 'id' | 'scheduledAt'>,
+  ) {
+    const event = this.getBackgroundEventWithId(backgroundEventWithoutId);
+    event.scheduledAt = new Date().toISOString();
```

Comment:
```suggestion
    const event = { ...backgroundEventWithoutId, id: nanoid(), scheduledAt: new Date().toISOString() }
```

Any reason to not just do this?